firebase-vertexai/src/main/kotlin/com/google/firebase/vertexai/common/util/kotlin.kt

```diff
@@ -39,3 +46,55 @@ internal fun StringBuilder.removeLast(): StringBuilder =
  * ```
  */
 internal inline fun <reified T : Annotation> Field.getAnnotation() = getAnnotation(T::class.java)
+
+/**
+ * Collects bytes from this flow and doesn't emit them back until [minSize] is reached.
+ *
+ * For example:
+ * ```
+ * val byteArr = flowOf(byteArrayOf(1), byteArrayOf(2, 3, 4), byteArrayOf(5, 6, 7, 8))
+ * val expectedResult = listOf(byteArrayOf(1, 2, 3, 4), byteArrayOf( 5, 6, 7, 8))
+ *
+ * byteArr.accumulateUntil(4).toList() shouldContainExactly expectedResult
+ * ```
+ *
+ * @param minSize The minimum about of bytes the array should have before being sent down-stream
+ * @param emitLeftOvers If the flow completes and there are bytes left over that don't meet the
+ * [minSize], send them anyways.
+ */
+internal fun Flow<ByteArray>.accumulateUntil(
+  minSize: Int,
+  emitLeftOvers: Boolean = false
+): Flow<ByteArray> = flow {
+  val buffer = mutableListOf<Byte>()
```

Comment:
what about using `ByteArrayOutputStreams`? They can be re-use already allocated space.